src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -2183,6 +2222,41 @@ struct Simplex_tree_options_full_featured {
   static const bool store_filtration = true;
   static const bool contiguous_vertices = false;
   static const bool link_nodes_by_label = false;
+  static const bool simplex_handle_strong_validity = false;
+};
+
+/** Model of SimplexTreeOptions, same as `Simplex_tree_options_full_featured`
+ *  but the possibility of much bigger keys and in particular of negative keys.
+ *
+ * Maximum number of simplices to compute persistence is <CODE>std::numeric_limits<std::int64_t>::max()</CODE>
+ * (way enough simplices). */
+struct Simplex_tree_options_wide_indexation {
```

Comment:
All those new `Simplex_tree_options_*` don't really seem related to this PR. I am not convinced we need a ton of *default* options to choose from, I'd rather users derive from one of very few *defaults* and customize the property they want. Even for the purpose of testing, we can define an options class in the test file.